pkg/gofr/cron.go

```diff
@@ -194,39 +194,45 @@ func parseRange(s string, minValue, maxValue int) (map[int]struct{}, error) {
 	r := make(map[int]struct{})
 	parts := strings.Split(s, ",")
 
-	for _, x := range parts {
-		rng := matchRange.FindStringSubmatch(x)
-
-		if rng == nil {
-			i, err := strconv.Atoi(x)
-			if err != nil {
-				return nil, errParsing{x, s}
-			}
-
-			if i < minValue || i > maxValue {
-				return nil, errOutOfRange{i, s, minValue, maxValue}
-			}
+	for _, part := range parts {
+		if err := parseSingleOrRange(part, minValue, maxValue, r); err != nil {
+			return nil, err
+		}
+	}
 
-			r[i] = struct{}{}
+	if len(r) == 0 {
+		return nil, errParsing{invalidPart: s}
+	}
 
-			continue
-		}
+	return r, nil
+}
 
+func parseSingleOrRange(part string, minValue, maxValue int, r map[int]struct{}) error {
+	if rng := matchRange.FindStringSubmatch(part); rng != nil {
 		localMin, _ := strconv.Atoi(rng[1])
 		localMax, _ := strconv.Atoi(rng[2])
 
 		if localMin < minValue || localMax > maxValue {
-			return nil, errOutOfRange{x, s, minValue, maxValue}
+			return errOutOfRange{part, part, minValue, maxValue}
 		}
 
-		r = getDefaultJobField(localMin, localMax, 1)
-	}
+		for i := localMin; i <= localMax; i++ {
```

Comment:
Can we use the `getDefaultJobField` function which does the same

Reply:
we can however i dont think so this i provides much more clarity and optimisation